site/zenodo_rdm/legacy/requests/community_manage_record.py

```diff
@@ -0,0 +1,211 @@
+
+
+#
+# Actions
+#
+class SubmitAction(actions.SubmitAction):
+    """Submit action."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        self.request["title"] = "Communities manage legacy records"
+
+        # example: "May 11, 2024"
+        expires_at = self.request.expires_at.strftime("%B %d, %Y")
+        self.request["description"] = (
+            "<h4>Some of your records, that are going through migration process are part "
+            "of the communities that don't belong to you.</br>Accept this request to keep the old "
+            "behaviour and <b>allow community curators</b> to manage (edit, create new version, add to "
+            "another community, etc.) corresponding record. </br>In case of declining this "
+            "request all your legacy records will be <b>removed from all communities</b> "
+            "that you are not an owner of. </br></br>If you do not perform any action by "
+            f"<b>{expires_at}</b>, the permission for community curators to manage the record "
+            "will automatically be fully granted.</h4>"
+        )
+
+        super().execute(identity, uow)
+
+
+class AcceptAction(actions.AcceptAction):
+    """Accept action."""
+
+    def execute(self, identity, uow):
+        """Grant permission to manage all legacy records of a user to all the communities."""
+        legacy_records = _get_legacy_records_by_user()
+
+        for hit in legacy_records.hits:
+            # remove flag from record parent, permissions logic will do the rest
+            record = _resolve_record(hit)
+            _remove_permission_flag(record, uow)
+
+        comment = (
+            "You accepted the request. The community curators "
+            "can now manage all of your legacy records."
+        )
+        _create_comment(self.request, comment, uow)
+
+        super().execute(identity, uow)
+
+
+class DeclineAction(actions.DeclineAction):
+    """Decline action."""
+
+    def execute(self, identity, uow):
+        """Deny access to manage legacy records for community curators."""
+        legacy_records = _get_legacy_records_by_user()
+
+        for legacy_record in legacy_records.hits:
+            record = _resolve_record(legacy_record)
+            communities = (
+                legacy_record["parent"].get("communities", None).get("ids", None)
+            )
+            if communities is not None:
+                not_my_communities = []
+                for community_id in communities:
+                    com_members = current_communities.service.members.search(
+                        system_identity,
+                        community_id,
+                        extra_filter=dsl.query.Bool(
+                            "must",
+                            must=[
+                                dsl.Q("term", **{"role": "owner"}),
+                                ~dsl.Q("term", **{"user_id": current_user.id}),
```

Comment:
Got one more question during testing: in case a community has multiple owners: me and not me, is the record still removed from it?

Reply:
If the record owner is also an owner of the community, the record should not be removed (though see my point above also about multiple members)